src/main/java/com/gildedrose/RedWine.java

```diff
@@ -0,0 +1,39 @@
+package com.gildedrose;
+
+public class RedWine extends DefaultItem {
+
+    private final float redWineQualityDegradationPastSellIn;
+    private final float redWineQualityDegradation;
+    private final Item item;
+    private float preciseQuality;
+
+    public RedWine(Item item) {
+        super(item);
+        this.redWineQualityDegradation = super.qualityDegradation/3.0f;
+        this.redWineQualityDegradationPastSellIn = this.redWineQualityDegradation * 2.0f;
+        this.item = item;
+        this.preciseQuality = item.quality;
+    }
+
+    public void age() {
+        item.quality = updateQuality();
+        item.sellIn = updateSellIn();
+    }
+
+    private int updateQuality() {
+        if(item.sellIn <= 0) {
+            preciseQuality -= redWineQualityDegradationPastSellIn;
+        }
+        else {
+            preciseQuality -= redWineQualityDegradation;
+        }
+
+        int qualityToReturn = Math.round(preciseQuality);
+
+        return Math.max(qualityToReturn, 0);
+    }
+
+    private int updateSellIn() {
```

Comment:
Question: If we're extending DefaultItem we can probably delegate to the parent class for this implementation?  Is this true for the other items that extend DefaultItem?

Reply:
Correct. I'm not even sure you need to override the method in the subclass if you're going to use the parent class implementation.